### ISSUE TITLE
every route is added with prefix length /32 to the kernel routing table

### ISSUE THREAD

**Issue by mstorck** [hessu/rip44d#7]
rip44d v1.2 adds every route with a prefix length of /32.

A few examples of logged $cmd:

```
/sbin/ip route add 44.161.0.0/32 via 141.75.245.225 dev tunl0 window 840 onlink  table 44
/sbin/ip route add 44.129.0.0/32 via 118.22.1.194 dev tunl0 window 840 onlink  table 44
/sbin/ip route add 44.147.0.0/32 via 121.99.232.227 dev tunl0 window 840 onlink  table 44
/sbin/ip route add 44.183.0.0/32 via 130.208.168.63 dev tunl0 window 840 onlink  table 44
/sbin/ip route add 44.94.0.0/32 via 141.224.128.8 dev tunl0 window 840 onlink  table 44
```

All these routes should be /16

I have already confirmed that the RIP message contains the correct netmask:

```
IP Address: 44.161.0.0, Metric: 1
    Address Family: IP (2)
    Route Tag: 4
    IP Address: 44.161.0.0 (44.161.0.0)
    Netmask: 255.255.0.0 (255.255.0.0)
    Next Hop: 141.75.245.225 (141.75.245.225)
    Metric: 1
```

OS: Debian Squeeze (6.0.7) for AMD64
rip44d: 1.2
kernel: 2.6.32-5-amd64
perl: v5.10.1 (*) built for x86_64-linux-gnu-thread-multi

**Comment by hessu** [comment]
Hi, can you post your rip44d script somewhere for download? I could take a look at its configuration and run it myself with your settings to see how it behaves.

Are you running the mainline version from https://github.com/hessu/rip44d or a modified branch?

**Comment by mstorck** [comment]
I'm running the unmodified version from Github (https://github.com/hessu/rip44d/blob/master/rip44d). As I was hunting the source of those /32 routes, I only a few "do_syslog(...);" to see what's going on

This is how I call the script:

```
/usr/local/sbin/rip44d_1.2 -a 176.9.140.87 -p fakePassword4GitHub -t 44 -s -v > /dev/null &
```

I can give you access to the server if you want.

**Comment by hessu** [comment]
Hi, I pushed a fix for the bug in SHA: b7bdcfba2a2a9aca230 - can you try the current version out and verify it works for you? I'm pretty sure it will.

**Comment by mstorck** [comment]
I can confirm the fix is working correctly: 

# ip route list table 44

default via x.y.66.251 dev tunl0 onlink 
44.0.0.1 via x.y.66.251 dev tunl0 onlink  window 840
44.2.1.32/29 via x.y.161.185 dev tunl0 onlink  window 840
44.2.8.180/30 via x.y.172.252 dev tunl0 onlink  window 840
44.2.10.0/29 via x.y.72.52 dev tunl0 onlink  window 840
...

Thanks alot!
